src/lib/viz/source/Source.ts

```diff
@@ -60,6 +60,11 @@ export abstract class Source extends WithEvents {
     this.registerAvailableEvents(['filterChange']);
   }
 
+  // eslint-disable-next-line class-methods-use-this
+  public isEmpty() {
```

Comment:
Why not an abstract method? I find strange to see an implementation in the interface...

Reply:
I wanted to add a _default_ implementation to avoid having to add the method in every source. But I reconsidered your suggestion and I agree that every source should implement its own method.